src/python_testing/TC_ICDM_5_1.py

```diff
@@ -0,0 +1,206 @@
+        print(services)
+        asserts.assert_true(mdns_discovery.MdnsServiceType.OPERATIONAL.value in services,
+                            "Could not find ICD operational service information.")
+
+        for service in services[mdns_discovery.MdnsServiceType.OPERATIONAL.value]:
+
+            # Generate hex for the compressedFabric and NodeId
+            serviceName = service.instance_name.split("-")
+            compressedFabric = hex(int(serviceName[0], 16))
+            nodeId = hex(int(serviceName[1], 16))
+
+            # Find the correct MdnsServiceEntry
+            if localCompressedNodeId == compressedFabric and dutNodeId == nodeId:
+                icdTxtRecord = service.txt_record['ICD']
+
+        return icdTxtRecord
+
+    #
+    # Test Harness Helpers
+    #
+
+    def desc_TC_ICDM_5_1(self) -> str:
+        """Returns a description of this test"""
+        return "[TC-ICDM-5.1] Operating Mode with DUT as Server"
+
+    def steps_TC_ICDM_5_1(self) -> list[TestStep]:
+        steps = [
+            TestStep(0, "Commissioning, already done", is_commissioning=True),
+            TestStep(1, "TH reads from the DUT the RegisteredClients attribute"),
+            TestStep("2a", "TH reads from the DUT the OperatingMode attribute."),
+            TestStep("2b", "Verify that the ICD DNS-SD TXT key is present."),
+            TestStep("3a", "TH sends RegisterClient command."),
+            TestStep("3b", "TH reads from the DUT the OperatingMode attribute."),
+            TestStep("3c", "Verify that mDNS is advertising ICD key."),
+            TestStep(4, "TH sends UnregisterClient command with CheckInNodeID1."),
+            TestStep("5a", "TH reads from the DUT the OperatingMode attribute."),
+            TestStep("5b", "Verify that the ICD DNS-SD TXT key is present."),
+        ]
+        return steps
+
+    def pics_TC_ICDM_5_1(self) -> list[str]:
+        """ This function returns a list of PICS for this test case that must be True for the test to be run"""
+        pics = [
+            "ICDM.S",
+            "ICDM.S.F02",
+        ]
+        return pics
+
+    #
+    # ICDM 5.1 Test Body
+    #
+
+    @async_test_body
+    async def test_TC_ICDM_5_1(self):
+
+        # Commissioning
+        self.step(0)
+
+        try:
+            self.step(1)
+            registeredClients = await self._read_icdm_attribute_expect_success(
+                attributes.RegisteredClients)
+
+            for client in registeredClients:
+                try:
+                    await self._send_single_icdm_command(commands.UnregisterClient(checkInNodeID=client.checkInNodeID))
+                except InteractionModelError as e:
+                    asserts.assert_equal(
+                        e.status, Status.Success, "Unexpected error returned")
+
+            self.step("2a")
+            operatingMode = await self._read_icdm_attribute_expect_success(attributes.OperatingMode)
+            asserts.assert_equal(operatingMode, modes.kSit.value)
```

Comment:
operatingMode seems to be already of type modes. No need for `.value` usage here or in the other places.